### PR TITLE
Factorio 2.0 / Space Age

### PR DESCRIPTION
**Pull request by SWeini**
Contains the changes necessary to work with Factorio 2.0, as well as some additions:
- Quality modules can be assigned like other modules, but they don't have an effect yet
- Rudimentary support for spoilage
- Rudimentary support for agricultural tower

Many features are still missing, but it's enough to start doing some calculations.
For further details, please see the checklists in the linked issues.

### CODE REVIEW

**Comment by shpaass** [comment]
Some of the tests don't pass, but I don't expect you to fix them for this transition. I think we'll fix them in a separate ticket.

**Comment by SWeini** [comment]
> Some of the tests don't pass, but I don't expect you to fix them now. I think we'll fix them in a separate ticket later.

it wasn't too bad
but I butchered the test for the various boiler fluid temperatures

**Comment by shpaass** [comment]
> I butchered the test for the various boiler fluid temperatures

I think we'll need to look into the boiler situation after pY decides what to do about boilers and liquid temperatures. 
Aside from that, the only thing that comes to mind is whether or not boiler works for vanilla. It would be good to have it correct, but it's not critical.

**Comment by SWeini** [comment]
> the only thing that comes to mind is whether or not boiler works for vanilla.
I made the current implementation work correctly for vanilla, which only uses one water temperature

**Comment by shpaass** [comment]
Then we're good. We'll have to address the pY temperatures once everything settles, I think.

File: Yafc.Parser/Data/FactorioDataDeserializer.cs [hunk 1]
```diff
@@ -333,15 +333,16 @@ private static float ParseEnergy(string? energy) {
             float energyBase = float.Parse(energy[..^2]);
 
             switch (energyMul) {
-                case 'k':
-                case 'K': return energyBase * 1e-3f;
+                case 'k': return energyBase * 1e-3f;
```
Comment (shpaass) [inline Yafc.Parser/Data/FactorioDataDeserializer.cs:327]: You removed the `K` case because it's no longer used?

**Comment by shpaass** [comment]
Squashed the commit of changing the Factorio version to 1.2 with changing it to 2.0. Did the same with type-6 and type-7. Added commit-prefixes,

**Review by shpaass — APPROVED** [review]

File: Yafc.Parser/Data/FactorioDataDeserializer.cs (hunk 1 above)
Comment (SWeini) [reply Yafc.Parser/Data/FactorioDataDeserializer.cs:327]: no longer supported

changelog excerpt:
> Added 'R' (ronna) and 'Q' (quetta) SI prefixes.
> Removed 'K' from allowed SI prefixes - use 'k' instead.

**Pull request merged**